File: 프로그래머스/1/388351. 유연근무제/유연근무제.c
#include <stdio.h>
#include <stdbool.h>

int solution(int schedules[], size_t schedules_len, int** timelogs, size_t timelogs_rows, size_t timelogs_cols, int startday) {
    int answer = 0;

    for (int i = 0; i < schedules_len; i++) {
        int currentDay = startday;
        bool success = true;

        for (int j = 0; j < timelogs_cols; j++) {
            if (currentDay > 7) currentDay = 1;

            if (currentDay == 6 || currentDay == 7) {
                currentDay++;
                continue;
            }

            int wishHour = schedules[i] / 100;
            int wishMinute = schedules[i] % 100;
            int allowedTime = wishHour * 100 + (wishMinute + 10);

            if (allowedTime % 100 >= 60) {
                allowedTime += 40; 
            }

            int clockInTime = timelogs[i][j];
            if (clockInTime > allowedTime) {
                success = false;
                break;
            }

            currentDay++;
        }

        if (success) {
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
class Solution {
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = 0;
        int count = 0;
        for(int i=0; i<timelogs.length; i++){
            count = 0;
            for(int j=0; j<timelogs[0].length; j++){
                int hour = schedules[i] / 100;
                int minute = schedules[i] % 100;
                int savetime = hour * 100 + (minute + 10);
                if(savetime % 100 >= 60)savetime += 40;
                if(timelogs[i][j] <= savetime && startday < 6){
                    count++;
                }
                startday++;
                if(startday > 7)startday = 1;
            }
            if(count == 5)answer++;
        }
        return answer;
    }
}
